src/UrlParser.elm

```diff
@@ -50,22 +57,38 @@ type alias State value =
 
 {-| Parse a segment of the path as a `String`.
 
-    parsePath string location
-    -- /alice/  ==>  Just "alice"
-    -- /bob     ==>  Just "bob"
-    -- /42/     ==>  Just "42"
+    parseString string "/alice/"
+    --> Just "alice"
+
+    parseString string "/bob"
+    --> Just "bob"
+
+    parseString string "/42/"
+    --> Just "42"
+
+    parseString string "/"
+    --> Nothing
 -}
 string : Parser (String -> a) a
 string =
-  custom "STRING" Ok
+  custom "STRING" <|
+    \segment ->
+      if String.isEmpty segment then
```

Comment:
If the documentation was in error, I can remove this fix, and update the doc tests to reflect it. I.e.

```elm
parseString string "/"
--> Just ""
```